### PR TITLE
[AF-572] Implement validation to prevent the creation of v1 apps targeting v2 only locations

### PR DESCRIPTION
**Pull request by fvzzy**
:v:

/cc @zendesk/vegemite

### Description
Adds validation for v2 only locations in v1 apps

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-572

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -97,6 +97,8 @@ en:
               version.
             locations_cant_be_urls: App locations must not be URLs for this framework
               version.
+            invalid_v1_location: "%{invalid_locations} is an invalid location in framework
```
Comment (sandlerr) [inline config/locales/en.yml]: need to support one or many? https://github.com/zendesk/zendesk_apps_support/blob/e16fdfb6fe5fa4111845db82d27c5fd28d6ade1d/config/locales/en.yml#L10-L11

File: lib/zendesk_apps_support/manifest.rb [hunk 2]
```diff
@@ -80,8 +80,11 @@ def app_class_properties
     end
 
     def unknown_locations(host)
+      product = Product.find_by(name: host)
+
       if locations.key?(host)
-        locations[host].keys.uniq - Location::LOCATIONS_AVAILABLE.map(&:name)
+        product_locations = Location.where(product_code: product.code)
+        locations[host].keys.uniq - product_locations.map(&:name)
```
Comment (sandlerr) [inline lib/zendesk_apps_support/manifest.rb]: :+1: nice

File: lib/zendesk_apps_support/validations/manifest.rb [hunk 3]
```diff
@@ -178,6 +179,25 @@ def invalid_location_error(package)
           errors
         end
 
+        def invalid_v1_location(package)
+          if package.manifest.framework_version &&
```
Comment (sandlerr) [inline lib/zendesk_apps_support/validations/manifest.rb]: mind removing a layer of nesting by returning early?

```ruby
def invalid_v1_location(package)
  return unless package.manifest.framework_version && Gem::Version ...

  invalid_locations = []
  ...
end
```

File: lib/zendesk_apps_support/validations/manifest.rb [hunk 4]
```diff
@@ -178,6 +179,25 @@ def invalid_location_error(package)
...
+            Gem::Version.new(package.manifest.framework_version) < Gem::Version.new('2')
+
+            invalid_locations = []
```
Comment (sandlerr) [inline lib/zendesk_apps_support/validations/manifest.rb]: code golf: could write it as

```ruby
invalid_locations = package.manifest.location_options
  .map(&:location)
  .compact
  .select(&:v2_only)
  .map(&:name)
```

**Comment by sandlerr** [comment]
does setting iframe URLs in a v1 app also error out?

**Comment by adammw** [comment]
I might be overengineering here, but having a 'v2' and 'v1' variables seems to smell of repetition a bit to me, would it be worth having some whitelist approach in case we add more locations for future versions?

**Comment by injuhwang** [comment]
@zendesk/i18n

**Pull request merged**